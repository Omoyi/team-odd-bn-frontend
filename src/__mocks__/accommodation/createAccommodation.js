// create accommodation reducers
import {
  NEW_ACCOMMODATION_INPUT,
  NEW_ADDED_PHOTOS,
  OPEN_ROOMS_MODAL,
  NEW_ADDED_ROOMS,
  GET_CITIES_SUCCESS,
  CREATE_ACCOMMODATION_SUCCESS,
  CREATE_ACCOMMODATION_ERROR,
  SPINNER_STATUS,
} from '../../redux/actionTypes/accommodationActionTypes';

export const accommodationInputAction = {
  type: NEW_ACCOMMODATION_INPUT,
  payload: {
    name: 'Kilimanjaro',
    city: 3,
    description: 'Nice place to rest in the world',
    address: 'KG 208 st',
    imageUrls: ['me.jpg', 'you.png'],
    rooms: [
      {
        name: 'Kagera',
        roomType: 'Single',
      },
      {
        name: 'Kakum',
        roomType: 'Double',
      },
    ],
  },
};
export const photosInputAction = {
  type: NEW_ADDED_PHOTOS,
  photos: 'me.you',
};
export const openModalAction = {
  type: OPEN_ROOMS_MODAL,
  openRoomsModal: true,
};
export const addNewRoomsAction = {
  type: NEW_ADDED_ROOMS,
  rooms: 'Akagera',
};
export const getCitiesSuccessAction = {
  type: GET_CITIES_SUCCESS,
  retrievedCities: {
    message: 'Location retrieved',
    data: [{
      id: 1,
      city: 'Nairobi',
      createdAt: '2019-12-09T07:36:36.417Z',
      updatedAt: '2019-12-09T07:36:36.417Z',
    }],
  },
};

export const createAccommodation = {
  type: CREATE_ACCOMMODATION_SUCCESS,
  payload: {
    name: 'Mubarak Palace',
    cityId: 3,
    address: 'fghj',
    googleCoordinates: '-77, -56',
    imageUrls: ['l.jpg', 'k.jpg'],
    description:
             ' Queens palace is the best accomodation facility you could ever get',
    rooms: [
      { name: '12', roomType: 'single' },
      { name: '13', roomType: 'single' },
      { name: '14', roomType: 'single' },
    ],
  },
};
export const createAccommodationError = {
  type: CREATE_ACCOMMODATION_ERROR,
  status: 'error',
  error: 'You are not a travel administrator',
};

export const spinnerStatus = {
  type: SPINNER_STATUS,
  spinner: false,
};

// create accommodation action

export const retrievedCitiesAction = [
  {
    spinner: false,
    type: SPINNER_STATUS,
  },
  {
    type: GET_CITIES_SUCCESS,
    retrievedCities: {
      message: 'Location retrieved',
      data: [
        {
          id: 1,
          city: 'Kampala',
          createdAt: '2019-12-09T07:36:36.417Z',
          updatedAt: '2019-12-09T07:36:36.417Z',
        },
      ],
    },
  },
  {
    type: CREATE_ACCOMMODATION_ERROR,
    status: 'success',
    error: {},
  },
];
export const retrievedCitiesApi = {
  status: 200,
  response: {
    message: 'Location retrieved',
    data: [
      {
        id: 1,
        city: 'Kampala',
        createdAt: '2019-12-09T07:36:36.417Z',
        updatedAt: '2019-12-09T07:36:36.417Z',
      },
    ],
  },
};

export const tokenErrorApiRequest = {
  status: 400,
  response: {
    message: 'JWT malformed',
  },
};

export const createAccommodationApi = {
  status: 201,
  response: {
    message: 'Accommodation created',
    data: {
      id: 24,
      userId: 15,
      name: "Jox's palace lo",
      cityId: 3,
      address: 'fghj',
      description:
               ' Queens palace is the best accomodation facility you could ever get',
      googleCoordinates: '-77, -56',
      updatedAt: '2020-01-15T18:17:26.264Z',
      createdAt: '2020-01-15T18:17:26.264Z',
      imageUrls: [
        {
          id: 34,
          accommodationId: 24,
          imageUrl: 'l.jpg',
          updatedAt: '2020-01-15T18:17:26.277Z',
          createdAt: '2020-01-15T18:17:26.277Z',
        },
      ],
      rooms: [
        {
          id: 46,
          name: 'Mugwizabatware',
          roomType: 'single',
          accommodationId: 24,
          updatedAt: '2020-01-15T18:17:26.318Z',
          createdAt: '2020-01-15T18:17:26.318Z',
        },
      ],
    },
  },
};
export const createAccommodationMockAction = [
  {
    spinner: false,
    type: SPINNER_STATUS,
  },
  {
    type: CREATE_ACCOMMODATION_SUCCESS,
    payload: {
      message: 'Accommodation created',
      data: {
        id: 24,
        userId: 15,
        name: "Jox's palace lo",
        cityId: 3,
        address: 'fghj',
        description:
                 ' Queens palace is the best accomodation facility you could ever get',
        googleCoordinates: '-77, -56',
        updatedAt: '2020-01-15T18:17:26.264Z',
        createdAt: '2020-01-15T18:17:26.264Z',
        imageUrls: [
          {
            id: 34,
            accommodationId: 24,
            imageUrl: 'l.jpg',
            updatedAt: '2020-01-15T18:17:26.277Z',
            createdAt: '2020-01-15T18:17:26.277Z',
          },
        ],
        rooms: [
          {
            id: 46,
            name: 'Mugwizabatware',
            roomType: 'single',
            accommodationId: 24,
            updatedAt: '2020-01-15T18:17:26.318Z',
            createdAt: '2020-01-15T18:17:26.318Z',
          },
        ],
      },
    },
  },
  {
    type: CREATE_ACCOMMODATION_ERROR,
    status: 'success',
    error: {},
  },
];
export const createAccommodationMockError = [
  {
    spinner: false,
    type: SPINNER_STATUS,
  },
  {
    type: CREATE_ACCOMMODATION_ERROR,
    status: 'error',
    error: {
      message: 'You are not a travel admnistrator',
    },
  },
];
export const createAccommodationMockApiError = {

  status: 401,
  response: {
    data: {
      message: 'You are not a travel admnistrator',
    },
  },
};
export const createAccommodationMockApiErrors = {
  status: 401,
  response: {
    data: {
      message: [
        'You are not a travel admnistrator',
        'imageURL should be valid',
      ],
    },
  },
};
export const createAccommodationMockErrors = [
  {
    spinner: false,
    type: SPINNER_STATUS,
  },
  {
    type: CREATE_ACCOMMODATION_ERROR,
    status: 'error',
    error: {
      message: [
        'You are not a travel admnistrator',
        'imageURL should be valid',
      ],
    },
  },
];
// Props
export const createAccommodationProps = {
  stateObject: {
    accommodation: {
      createAccommodation: {
        spinner: false,
        openRoomsModal: false,
        retrievedCities: {
          data: [
            {
              id: 1,
              city: 'Kampala',
              createdAt: '2019-12-09T07:36:36.417Z',
              updatedAt: '2019-12-09T07:36:36.417Z',
            },
            {
              id: 2,
              city: 'Kigali',
              createdAt: '2019-12-09T07:36:36.417Z',
              updatedAt: '2019-12-09T07:36:36.417Z',
            },
          ],
        },
        payload: {
          message: 'Accommodation posted successfully',
        },
      },
    },
  },
  createAccommodationAction: jest.fn(),
  newAccommodationInput: jest.fn(),
  newAccommodationPhotos: jest.fn(),
  roomsModalAction: jest.fn(),
  getCitiesAction: jest.fn(),
  updateSpinnerStatus: jest.fn(),
};
export const modalSpinnerProps = {
  stateObject: {
    accommodation: {
      createAccommodation: {
        spinner: true,
        openRoomsModal: true,
        retrievedCities: {
          data: [
            {
              id: 1,
              city: 'Kampala',
              createdAt: '2019-12-09T07:36:36.417Z',
              updatedAt: '2019-12-09T07:36:36.417Z',
            },
            {
              id: 2,
              city: 'Kigali',
              createdAt: '2019-12-09T07:36:36.417Z',
              updatedAt: '2019-12-09T07:36:36.417Z',
            },
          ],
        },
      },
    },
  },
  createAccommodationAction: jest.fn(),
  newAccommodationInput: jest.fn(),
  newAccommodationPhotos: jest.fn(),
  roomsModalAction: jest.fn(),
  getCitiesAction: jest.fn(),
  updateSpinnerStatus: jest.fn(),
};
export const mainState = {
  accommodation: {
    createAccommodation: {
      rooms: [],
      photos: [],
    },
  },
};

export const modalProps = {
  appendRow: jest.fn(),
  addNewRoomsAction: jest.fn(),
  roomsModalAction: jest.fn(),
};
