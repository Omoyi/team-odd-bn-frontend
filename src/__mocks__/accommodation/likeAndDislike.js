const likeAndDislikeMockData = {
  likeInput: {
    status: 200,
    response: {
      message: 'like successfully',
      payload: [
        {
          LikesDislikes: {
            dislikeCounter: 1,
            updatedAt: '2020-01-24T14:29:50.618Z',
            userId: 15,
          },
        },
      ],
    },
  },

  likeExpectedResponse: [
    {
      type: 'LIKES_SUCCESS',
      payload: {
        message: 'like successfully',
        payload: [{
          LikesDislikes: {
            dislikeCounter: 1,
            updatedAt: '2020-01-24T14:29:50.618Z',
            userId: 15,
          },
        }],
      },
    },
  ],

  dislikeInput: {
    status: 200,
    response: {
      message: 'dislike successfully',
      payload: [
        {
          LikesDislikes: {
            dislikeCounter: 1,
            updatedAt: '2020-01-24T14:29:50.618Z',
            userId: 15,
          },
        },
      ],
    },
  },
  dislikeExpectedResponse: [
    {
      type: 'DISLIKES_SUCCESS',
      payload: {
        message: 'dislike successfully',
        payload: [{
          LikesDislikes: {
            dislikeCounter: 1,
            updatedAt: '2020-01-24T14:29:50.618Z',
            userId: 15,
          },
        }],
      },
    },
  ],

};
export default likeAndDislikeMockData;
