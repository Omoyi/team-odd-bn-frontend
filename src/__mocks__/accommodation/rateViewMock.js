export const mapViewProps = {
  accommodationItem: {
    id: 45,
    userId: 15,
    name: 'Eric Palace',
    cityId: 2,
    address: 'Kigali',
    description: 'I am going there',
    googleCoordinates: '-22,-12',
    createdAt: '2020-01-24T14:29:50.618Z',
    updatedAt: '2020-01-24T14:29:50.618Z',
    accommodationRooms: [
      {
        id: 88,
        accommodationId: 45,
        name: 'Kackity',
        roomType: 'Double',
        createdAt: '2020-01-24T14:29:50.761Z',
        updatedAt: '2020-01-24T14:29:50.761Z',
      },
      {
        id: 89,
        accommodationId: 45,
        name: 'Serena',
        roomType: 'Single',
        createdAt: '2020-01-24T14:29:50.762Z',
        updatedAt: '2020-01-24T14:29:50.762Z',
      },
    ],
    imagesAccommodation: [
      {
        id: 62,
        accommodationId: 45,
        imageUrl: 'http://res.cloudinary.com/victorkarangwa4/image/upload/v1579876184/Screen_Shot_2019-12-19_at_10.33.23_gnuiwa.png',
        createdAt: '2020-01-24T14:29:50.665Z',
        updatedAt: '2020-01-24T14:29:50.665Z',
      },
      {
        id: 61,
        accommodationId: 45,
        imageUrl: 'http://res.cloudinary.com/victorkarangwa4/image/upload/v1579876134/34_bdjnqy.jpg',
        createdAt: '2020-01-24T14:29:50.657Z',
        updatedAt: '2020-01-24T14:29:50.657Z',
      },
    ],
  },
  popUpAction: jest.fn(),
  prevProps: {
    accommodationItem: {
      id: 47,
      userId: 15,
      name: 'Eric Palace',
      cityId: 2,
      address: 'Kigali',
      description: 'I am going there',
      googleCoordinates: '-22,-12',
      createdAt: '2020-01-24T14:29:50.618Z',
      updatedAt: '2020-01-24T14:29:50.618Z',
      accommodationRooms: [
        {
          id: 88,
          accommodationId: 45,
          name: 'Kackity',
          roomType: 'Double',
          createdAt: '2020-01-24T14:29:50.761Z',
          updatedAt: '2020-01-24T14:29:50.761Z',
        },
        {
          id: 89,
          accommodationId: 45,
          name: 'Serena',
          roomType: 'Single',
          createdAt: '2020-01-24T14:29:50.762Z',
          updatedAt: '2020-01-24T14:29:50.762Z',
        },
      ],
      imagesAccommodation: [
        {
          id: 62,
          accommodationId: 45,
          imageUrl: 'http://res.cloudinary.com/victorkarangwa4/image/upload/v1579876184/Screen_Shot_2019-12-19_at_10.33.23_gnuiwa.png',
          createdAt: '2020-01-24T14:29:50.665Z',
          updatedAt: '2020-01-24T14:29:50.665Z',
        },
        {
          id: 61,
          accommodationId: 45,
          imageUrl: 'http://res.cloudinary.com/victorkarangwa4/image/upload/v1579876134/34_bdjnqy.jpg',
          createdAt: '2020-01-24T14:29:50.657Z',
          updatedAt: '2020-01-24T14:29:50.657Z',
        },
      ],
    },
  },
};

export const viewPortMapMock = {
  width: 0,
  height: 0,
  latitude: -1.9500725,
  longitude: 30.0846178,
  zoom: 15,
  bearing: 0,
  pitch: 0,
  altitude: 1.5,
  maxZoom: 24,
  minZoom: 0,
  maxPitch: 60,
  minPitch: 0,
  transitionDuration: 0,
};


export const cityInfoProps = {
  info: {
    name: 'John',
    address: 'Kigali',
    image: 'user.jpg',
  },
};


export const pinMockPropsData = {
  data: [
    {
      address: 'Kigali',
      name: 'Eric Palace',
      image: 'https://images.unsplash.com/photo-1559599238-30879â€¦ib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
      latitude: -1.9500725,
      longitude: 30.0846178,
    },
  ],
  onClick: jest.fn(),
};
