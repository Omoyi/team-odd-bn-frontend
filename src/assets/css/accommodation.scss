@import './App.scss';
@import './base.scss';

.center-container {
  display: flex;
  flex-direction: column;
  margin: 10px auto;
  width: 620px;
  height: 750px;
  .page-header {
    margin-bottom: 10px;
    margin-top: 10px;
  }
  .accommodation-container {
    width: 620px;
    height: 750px;
    border: 1px solid $logo-color;
    .inputbox {
      display: flex;
      flex-direction: column;
      .name {
        margin: 10px auto;
      }
      .city {
        margin: 10px auto;
      }
      .googleCoordinates {
        margin: 10px auto;
      }
      .address {
        margin: 10px auto;
      }
      .description {
        margin: 30px auto 10px auto;
      }
      .aligned-container {
        margin: 10px auto;
        width: 435px;
      }
      .upload-btn {
        margin-top: 10px;
        .uploaded-photos {
          margin-top: 14px;
        }
      }
      .btn-long {
        width: 140px;
        margin-left: 1px !important;
        @extend %btn-long;
        text-align: center;
      }
      button {
        width: 160px;
        height: 40px;
        font-size: 16px;
        background-color: #c4c4c4;
      }
      input {
        width: 435px;
        height: 45px;
        margin-top: 10px;
        font-size: 15px;
        border: 1px solid #5c5c5c;
        border-radius: 2px;
        padding: 0px 0px 0px 10px!important;
        margin-right: 0px !important;
        font-family: muli;
      }
      select {
        width: 435px;
        height: 30px;
        margin-top: 10px;
        font-size: 15px;
        border: 1px solid #5c5c5c;
        border-radius: 2px;
        margin-right: 0px !important;
                padding: 0px 0px 0px 10px!important;
      }
      textarea {
        width: 435px;
        height: 60px;
        margin-top: 10px;
          font-size: 14px;
        border: 1px solid #5c5c5c;
        border-radius: 2px;
                padding: 0px 0px 0px 10px!important;
      }
    }
  }
}

@media screen and (min-width: 1024px) and (max-width: 1223px) {
}
@media screen and (min-width: 768px) and (max-width: 1000px) {
}

@media screen and (min-width: 324px) and (max-width: 1024px) {
}
@media screen and (max-width: 768px) {
  .center-container {
    display: flex;
    flex-direction: column;
    margin: 10px auto;
    width: 100%;
    height: 700px;
    .page-header {
      margin-bottom: 10px;
      margin-left: 5px;
    }
    .accommodation-container {
      width: 100%;
      height: 700px;
      border: 1px solid $logo-color;
      .inputbox {
        display: flex;
        flex-direction: column;
        margin-top: 20px;
        .name {
          margin: auto;
        }
        .city {
          margin: auto;
        }
        .googleCoordinates {
          margin: auto;
        }
        .address {
          margin: auto;
        }
        .description {
          margin-top: 40px !important;
          textarea {
            width: 230px;
                      
          }
        }
        .aligned-container {
          margin: 10px auto;
          width: 385px;
        }
        .upload-btn {
          margin-left: 6px;
        }
        .btn-long {
          margin-left: 5px !important;
        }
        button {
          width: 140px;
          height: 30px;
          background-color: #c4c4c4;
        }
        input {
          width: 100%;
          height: 45px;
          margin-top: 10px;
          font-size: 15px;
          border: 1px solid #5c5c5c;
          border-radius: 2px;
          margin-bottom: 5px;
          font-family: muli;
        }
        select {
          width: 235px;
          height: 30px;
          margin-top: 10px;
          font-size: 10px;
          border: 1px solid #5c5c5c;
          border-radius: 2px;
                    margin-bottom: 10px;
        }

      }
    }
  }
}
@media (max-width: 768px) and (min-width: 360px) {
  .center-container {
    display: flex;
    flex-direction: column;
    margin: 10px auto;
    width: 100%;
    height: 700px;
    .page-header {
      margin-bottom: 10px;
      margin-left: 5px;
    }
    .accommodation-container {
      width: 100%;
      height: 700px;
      border: 1px solid $logo-color;
      .inputbox {
        display: flex;
        flex-direction: column;
        margin-top: 20px;
        .name {
          margin: auto;
        }
        .city {
          margin: auto;
        }
        .googleCoordinates {
          margin: auto;
        }
        .address {
          margin: auto;
        }
        .description {
          margin: auto;
          textarea {
            width: 380px;
          }
        }
        input {
          width: 380px;
          height: 45px;
          margin-top: 10px;
          font-size: 15px;
          border: 1px solid #5c5c5c;
          border-radius: 2px;
          padding-left: 100px;
          font-family: muli;
        }
        select {
          width: 380px;
          height: 30px;
          margin-top: 10px;
          font-size: 10px;
          border: 1px solid #5c5c5c;
          border-radius: 2px;
        }
      }
    }
  }
}
@media screen and (max-width: 380px){
  .center-container {
    display: flex;
    flex-direction: column;
    margin: 10px auto;
    width: 100%;
    height: 700px;
    .page-header {
      margin-bottom: 10px;
      margin-left: 5px;
    }
    .accommodation-container {
      width: 100%;
      height: 700px;
      border: 1px solid $logo-color;
      .inputbox {
                margin-top: 30px;
        display: flex;
        flex-direction: column;
        margin-top: 20px;
        .name {
          margin: auto;
        }
        .city {
          margin: auto;
        }
        .googleCoordinates {
          margin: auto;
        }
        .address {
          margin: auto;
        }
        .description {
          margin: auto;
          textarea {
            width: 230px;
          }
        }
        .aligned-container {
          margin: 10px auto;
          width: 385px;
        }
        .upload-btn {
          margin-left: 6px;
        }
        .btn-long {
          margin-left: 5px !important;
        }
        button {
          width: 140px;
          height: 30px;
          background-color: #c4c4c4;
        }
        input {
          width: 100%;
          height: 45px;
          margin-top: 10px;
          font-size: 15px;
          border: 1px solid #5c5c5c;
          border-radius: 2px;
          font-family: muli;
        }
        select {
          width: 235px;
          height: 30px;
          margin-top: 10px;
          font-size: 10px;
          border: 1px solid #5c5c5c;
          border-radius: 2px;
        }
      }
    }
  }
}