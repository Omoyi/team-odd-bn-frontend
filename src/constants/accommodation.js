
export const createAccommodationList = [
  {
    input: 'Name',
    className: 'name',
    type: 'text',
    title: 'Name should be valid',
  },
  {
    input: 'Google coordinates',
    className: 'googleCoordinates',
    type: 'text',
    title: 'Name should be valid',
  },
  {
    input: 'Address',
    className: 'address',
    type: 'text',
    title: 'address should be valid',
  },
];
export const modalInput = [
  {
    input: 'Room name',
    className: 'room-name',
    type: 'text',
    title: 'Room name should be valid',
  },
];
