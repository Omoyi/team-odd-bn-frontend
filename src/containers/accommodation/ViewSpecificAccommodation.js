import React, { Component } from 'react';
import { connect } from 'react-redux';
import { ToastContainer } from 'react-toastify';
import { Redirect } from 'react-router-dom';
import '../../assets/css/accommodation/viewSpeciAccommodation.scss';
import Dashboard from '../../views/Dashboard/sidebar/index';
import AccommodationsRooms from '../../views/accommodation/AccommodationsRooms';
import AccommodationReviews from '../../views/accommodation/AccommodationReviews';
import ViewAccomodationDetails from '../../views/accommodation/ViewSpecificAccommodation';
import singleAccommodation from '../../redux/actions/singleAccommodation';
import Spinner from '../../components/Spinner';


export class ViewSpecificAccommodation extends Component {
  componentDidMount() {
    const { singleAccommodation, match: { params: { id } } } = this.props;
    singleAccommodation(id);
  }


  render() {
    const { accommodationItem, accommodationItemError } = this.props;

    return (
      <Dashboard>
        <ToastContainer />
        { accommodationItemError.length > 0 && <Redirect to="/accommodations" /> }
        { !accommodationItem ? <Spinner />
          : (
            <>
              <ViewAccomodationDetails />
              <AccommodationsRooms />
              <AccommodationReviews />
            </>
          )}
      </Dashboard>
    );
  }
}

export const mapStateToProps = (state) => {
  return {
    accommodationItem: state.specificAccommodation.singleAccommodation[0],
    accommodationItemError: state.specificAccommodation.singleAccommodationError,
  };
};

export const mapDispatchToprops = {
  singleAccommodation,
};

export default connect(mapStateToProps, mapDispatchToprops)(ViewSpecificAccommodation);
