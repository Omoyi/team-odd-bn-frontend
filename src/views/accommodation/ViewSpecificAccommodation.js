import React, { Component } from 'react';
import { connect } from 'react-redux';
import '../../assets/css/accommodation/viewSpeciAccommodation.scss';
import singleAccommodation from '../../redux/actions/singleAccommodation';
import PopUp from '../../components/PopUp';
import AccommodationMap from './AccommodationMap';
import popUpAction from '../../redux/actions/popUpAction';

export class ViewSpecificAccommodation extends Component {
  render() {
    const { accommodationItem, popUpAction,display } = this.props;
    return (
      <>
        <PopUp display={display.map} popUp={<AccommodationMap />} />
        <div className="accommodation-title">
          <h1>Accommodation details</h1>
        </div>
        <div className="accommodation-container-details">
          <div className="accommodation-details-data">
            <div className="main-accommodation-image">
              <img src={accommodationItem.imagesAccommodation[0].imageUrl} alt="main accommodation" />
            </div>
            <div className="main-accommodation-details">
              <h2 className="main-accommodation-details-title">{ accommodationItem.name }</h2>
              <section className="common-section city-section">
                <p className="common-section-name label-accommodation-details">City: </p>
                <p className="accommodation-details-single-data">{ accommodationItem.address }</p>
              </section>
              <section className="common-section room-section">
                <p className="common-section-name label-accommodation-details">Rooms: </p>
                <p className="accommodation-details-single-data">
                  { accommodationItem.accommodationRooms.length }
                  types of rooms
                </p>
              </section>
              <section className="common-section description-section">
                <p className="common-section-name label-accommodation-details">Description: </p>
                <p className="accommodation-details-single-data description-section-text">
                  { accommodationItem.description }
                </p>
              </section>
              <section className="common-section created-section">
                <p className="common-section-name label-accommodation-details">Opened at: </p>
                <p className="accommodation-details-single-data">
                  { new Date().toDateString(accommodationItem.createdAt) }
                </p>
              </section>
              <section className="common-section coordinates-section">
                <button
                  className="visit-map-button"
                  onClick={() => popUpAction({
                    map: 'flex',
                    currentPopUp: 'map',
                  })}
                  type="button"
                >
                  View on map
                </button>
              </section>
            </div>
          </div>
        </div>
      </>
    );
  }
}

export const mapStateToProps = (state) => ({
  accommodationItem: state.specificAccommodation.singleAccommodation[0],
  display: state.popUpsDisplay,
});

export const mapDispatchToprops = {
  singleAccommodation,
  popUpAction,
};

export default connect(mapStateToProps, mapDispatchToprops)(ViewSpecificAccommodation);
