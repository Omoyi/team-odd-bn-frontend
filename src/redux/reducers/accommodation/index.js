import { combineReducers } from 'redux';
import createAccommodation from './createAccommodationReducer';

export default combineReducers({
  createAccommodation,
});
