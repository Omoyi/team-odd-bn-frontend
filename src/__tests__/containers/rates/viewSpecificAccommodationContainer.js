/* eslint-disable import/no-duplicates */
import React from 'react';
import { shallow } from 'enzyme';
import { ViewSpecificAccommodation } from '../../../containers/accommodation/ViewSpecificAccommodation';
import {
  viewSpecificAccommodationMocks,
  noSpecificAccommodationMocks,
} from '../../../__mocks__/RatingMocks/BookingAccommodation';


describe('Expect to view specific accommodation', () => {
  it('Should render specific accommodation successfully', () => {
    const wrapper = shallow(<ViewSpecificAccommodation {...viewSpecificAccommodationMocks} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should not continue if no specific accommodation available', () => {
    const wrapper = shallow(<ViewSpecificAccommodation {...noSpecificAccommodationMocks} />);
    expect(wrapper).toMatchSnapshot();
  });
});
