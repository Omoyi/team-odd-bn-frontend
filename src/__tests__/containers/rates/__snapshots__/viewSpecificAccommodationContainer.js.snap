// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to view specific accommodation Should not continue if no specific accommodation available 1`] = `ShallowWrapper {}`;

exports[`Expect to view specific accommodation Should render specific accommodation successfully 1`] = `ShallowWrapper {}`;
